### PR TITLE
Update wrapper and example for grpc options API

### PR DESCRIPTION
**Pull request by Shalini-Subramanian**
### What does this Pull Request accomplish?

This Pull Request updates the `dut_communication_using_semi_device_control_nidigital.py` example to attach to the NI Digital gRPC session using the newly created API `get_grpc_session_options` .

### Why should this Pull Request be merged?

This Pull Request updates the example to access the NI Digital session via the NI gRPC Device Server using the newly created API `get_grpc_session_options`.

### What testing has been done?

Manually tested through Simulation mode.

### CODE REVIEW

File: examples/dut_communication_using_semi_device_control_nidigital.py [hunk 1]
```diff
@@ -44,18 +44,20 @@
 
     # Attach to the existing 657x gRPC session.
     # Get the resource name of the instrument session for NI 657x device.
-    resource_name = semi_device_control.get_instrument_session("NI 657x").ResourceName
+    grpc_session_options = semi_device_control.get_grpc_session_options("NI 657x")
```
Comment (SowmyaRavee) [inline examples/dut_communication_using_semi_device_control_nidigital.py:47]: Why `NI 657x` is hardcoded? What's the input parameter needed for `get_grpc_session_options`? Do we have to pass in the interface name?

File: examples/dut_communication_using_semi_device_control_nidigital.py [hunk 2]
```diff
@@ -44,18 +44,20 @@
...
     
     # Create an insecure gRPC channel to connect to the device server
-    device_server_channel = grpc.insecure_channel("localhost:31763")
+    device_server_channel = grpc.insecure_channel(grpc_session_options.Address + ":" + str(grpc_session_options.Port))
```
Comment (SowmyaRavee) [inline examples/dut_communication_using_semi_device_control_nidigital.py:50]: Should user be constructing this channel? Can we not create and give?

File: examples/dut_communication_using_semi_device_control_nidigital.py [hunk 3]
```diff
@@ -44,18 +44,20 @@
...
+
     grpc_option = nidigital.GrpcSessionOptions(
         grpc_channel= device_server_channel,
-        session_name=resource_name,
+        session_name=grpc_session_options.SessionName,
         initialization_behavior=nidigital.SessionInitializationBehavior.ATTACH_TO_SERVER_SESSION)
     
     # get the existing 657x instrument session.
+    # Session name will always be same as of the resource name, whenever the digital session is created from SDC
```
[the remark this answers is not in the document]
Comment (SowmyaRavee) [reply examples/dut_communication_using_semi_device_control_nidigital.py:58]: Oh did we decide to not return the resource name?

File: examples/dut_communication_using_semi_device_control_nidigital.py (hunk 1 above)
Comment (Shalini-Subramanian) [reply examples/dut_communication_using_semi_device_control_nidigital.py:47]: Yes, interface name is a required parameter for get_grpc_session_options()

File: examples/dut_communication_using_semi_device_control_nidigital.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (Shalini-Subramanian) [reply examples/dut_communication_using_semi_device_control_nidigital.py:58]: Resource name will be updated as discussed in the upcoming PR

File: examples/dut_communication_using_semi_device_control_nidigital.py (hunk 2 above)
Comment (Shalini-Subramanian) [reply examples/dut_communication_using_semi_device_control_nidigital.py:50]: Updated

File: examples/dut_communication_using_semi_device_control_nidigital.py (hunk 1 above)
Comment (SowmyaRavee) [reply examples/dut_communication_using_semi_device_control_nidigital.py:47]: Can we create a variable with this hardcoded value and use?

[the remark this answers is not in the document]
Comment (SowmyaRavee) [reply examples/dut_communication_using_semi_device_control_nidigital.py:47]: We have to add from this version

File: poetry.lock
Comment (SowmyaRavee) [inline poetry.lock:1]: We should not push lock file

We can delete this file in repo and add to .gitignore

Comment (GowthamSriniva) [reply poetry.lock:1]: we have a pipeline check which checks the consistency between the project.toml and poetry.toml file. [PR.yml](https://github.com/ni/nisemi-python/blob/1509ab8ef916a58d4f83bbef5c86e7a55c9255be/.github/workflows/PR.yml#L28)

Comment (SowmyaRavee) [reply poetry.lock:1]: I guess that check was added to ensure if the dependencies are properly mentioned and installed

File: nisdc/nisemidevicecontrol.py [hunk 4]
```diff
@@ -1170,6 +1171,24 @@ def get_instrument_session(self, interface_name):
...
             print("Exception occured at get instrument session")
             raise e
+        
+    def get_session_options(self, interface_name):
+        """Gets the grpc options for the instrument session.
+           Creates service server channel using the grpc options.
+
+            Return:
+                grpc_session_options { Object contains address {string}, port {int}, session_name{string} },
+                device_secure_channel { grpc.Channel }
```
Comment (SowmyaRavee) [inline nisdc/nisemidevicecontrol.py:1181]: Can we create a class structure to hold the grpc channel and Session name?

Comment (Shalini-Subramanian) [reply nisdc/nisemidevicecontrol.py:1185]: Updated

**Review by SowmyaRavee — APPROVED** [review]

**Review by GowthamSriniva — APPROVED** [review]

**Pull request merged**